### DISCUSSION TITLE
sbt scalaVersion not detected unless in same file as dependencies

### DISCUSSION THREAD

**Discussion by tmackenzie92** [renovatebot/renovate#19949] [Request Help]
Hello,

When setting up self-hosted Renovate recently I had an issue getting it to check for packages cross-compiled with different scala versions (e.g. named `my-dependency_2.12`, `my-dependency_2.13`, etc.) - it was only checking for the "root" dependency name.

Our project has a `build.sbt` at the project root and a `project/Dependencies.scala` file which looked something like this:

`build.sbt`
```
import Dependencies._

ThisBuild / scalaVersion := "2.13.8"

<use vals from Dependencies object>
```

`project/Dependencies.scala`
```
object Dependencies {
  val someDeps = Seq(
    "com.foo" %% "bar" % "1.0.0",
    ...
  )
  ...
}
```

In this situation, Renovate seemed only to search for new versions of `com.foo.bar` rather than `com.foo.bar_2.13`. I wondered if the scala version might only be detected for the file currently being parsed so I updated `Dependencies.scala` to contain
```
val thisBuildScalaVersion = "2.13.8"
ThisBuild / scalaVersion := thisBuildScalaVersion
```
and then 'build.sbt'
```
ThisBuild / scalaVersion := thisBuildScalaVersion
```
and this worked.

Is this how Renovate is intended to work, or should it detect scala version in any way that's valid for sbt?

I've reproduced this behaviour in a [minimal repo](https://github.com/tmackenzie92/renovate-test/) using the latest Renovate image as well as the original, much larger repo.

Thanks.

**Comment by rarkins** [comment]
@zharinov

**Reply by zharinov** [reply]
Our current dependency extraction doesn't perform any cross-file analysis

**Reply by zharinov** [reply]
We can start doing so, but I would like to think about the model of processing (i.e., in which order, some another concerns, etc)

**Comment by Butterneck** [comment]
Hello, I'm facing the exact same problem 👀
